### Leitura do arquivo .pkl

Leitura do arquivo onde foi salvo os previsores e as classes com seu devido pre processsamento


In [1]:
import pickle

In [2]:
with open('../../variaveis_preprocessadas/credit.pkl', 'rb') as f:
    previsores_treinamento, classe_treinamento, previsores_teste, classe_teste = pickle.load(f)

### Utilização do algoritmo KNN

In [8]:
from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier(n_neighbors=10, p=2)
knn.fit(previsores_treinamento, classe_treinamento)
previsoes = knn.predict(previsores_teste)

### Avaliação dos resultados do algoritmo KNN

In [4]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [9]:
score = accuracy_score(classe_teste, previsoes)
score

0.986

## Valores obtidos por parametros diferentes:

n_neighbours = 3; score = 0.984  <br>
n_neighbours = 5; score = 0.98   <br>
n_neighbours = 10; score = 0.986 <br>

In [10]:
confusion_matrix(classe_teste, previsoes)

array([[430,   2],
       [  5,  63]], dtype=int64)

In [11]:
print(classification_report(classe_teste, previsoes))

              precision    recall  f1-score   support

           0       0.99      1.00      0.99       432
           1       0.97      0.93      0.95        68

    accuracy                           0.99       500
   macro avg       0.98      0.96      0.97       500
weighted avg       0.99      0.99      0.99       500

